src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKYamlDSLParser.java

```diff
@@ -0,0 +1,150 @@
+		}
+		return camelComponentURI;
+	}
+
+	@Override
+	public CamelURIInstance createCamelURIInstance(TextDocumentItem textDocumentItem, Position position,
+			String camelComponentUri) {
+		CamelURIInstance uriInstance = new CamelURIInstance(repairLostEscapeChars(camelComponentUri), new YamlDSLModelHelper(getCorrespondingType(textDocumentItem, position.getLine())), textDocumentItem);
+		int start = getStartCharacterInDocumentOnLinePosition(textDocumentItem, position);
+		uriInstance.setStartPositionInDocument(new Position(position.getLine(), start));
+		uriInstance.setEndPositionInDocument(new Position(position.getLine(), start+repairLostEscapeChars(camelComponentUri).length()));
+		return uriInstance;
+	}
+
+	private String getNonNullValue(Object o) {
+		if (o != null) {
+			return o.toString();
+		}
+		return "";
+	}
+
+	private int getStartCharacterInDocumentOnLinePosition(TextDocumentItem textDocumentItem, Position position) {
+		String line = parserFileHelperUtil.getLine(textDocumentItem, position.getLine());
+		String uri = extractUriFromYamlData(line);
+		if (uri.length()==0) {
+			// empty uri
+			return findStartPositionOfEmptyURI(line);
+		}
+		uri = repairLostEscapeChars(uri);
+		return line.indexOf(uri);
+	}
+
+	@Override
+	public int getPositionInCamelURI(TextDocumentItem textDocumentItem, Position position) {
+		String line = parserFileHelperUtil.getLine(textDocumentItem, position.getLine());
+		return position.getCharacter() - findStartPositionOfEmptyURI(line);
+	}
+
+	private String repairLostEscapeChars(String line) {
+		String[] parts = line.split("\"");
+		String res = "";
+		for (int i = 0; i < parts.length; i++) {
+			if (i>0) {
+				res += "\\\"";
+			}
+			res += parts[i];
+		}
+		return res;
+	}
+
+	private int findStartPositionOfEmptyURI(String line) {
+		int startPos = line.indexOf(':')+2;
+		if (line.indexOf("'", startPos) != -1) {
+			// we use single quotes
+			return line.indexOf("'", startPos)+1;
+		} else if (line.indexOf('"', startPos) != -1) {
+			// we use double quotes
+			return line.indexOf('"', startPos)+1;
+		} else {
+			// we don't use any quotes
+			return startPos;
+		}
+	}
+
+	public String getCorrespondingType(TextDocumentItem textDocumentItem, int lineNumber) {
+		for (int lineNo = lineNumber; lineNo >=0; lineNo--) {
+			String tempLine = parserFileHelperUtil.getLine(textDocumentItem, lineNo);
+			Map<?, ?> data = parseYaml(tempLine);
+			if (data.containsKey(TO_KEY)) {
+				return "to";
+			} else if (data.containsKey(FROM_KEY)) {
+				return "from";
+			} else if (data.containsKey(REST_KEY)) {
+				return null;
+			}
+		}
+		return null;
+	}
+
+	private Map<?, ?> parseYaml(String line) {
+		Yaml yaml = new Yaml();
+		Object obj = yaml.load(line);
+		return extractMapFromYaml(obj);
+	}
+
+	private String extractUriFromYamlData(String line) {
+		Yaml yaml = new Yaml();
+		Object obj = yaml.load(line);
+		Map<?, ?> m = extractMapFromYaml(obj);
+		return getNonNullValue(m.values().toArray()[0]);
```

Comment:
Shoudl we check that m.values() is not empty before retrieving the first element?